ring/lifecycler.go

```diff
@@ -795,13 +805,13 @@ func (i *Lifecycler) changeState(ctx context.Context, state InstanceState) error
 
 func (i *Lifecycler) updateCounters(ringDesc *Desc) {
 	healthyInstancesCount := 0
-	zones := map[string]struct{}{}
+	zones := map[string]int{}
 
 	if ringDesc != nil {
 		now := time.Now()
 
 		for _, ingester := range ringDesc.Ingesters {
-			zones[ingester.Zone] = struct{}{}
+			zones[ingester.Zone]++
```

Comment:
I think for your purpose you should also count the healthy ones, like we do for `healthyInstancesCount` and then I would rename `InstancesInZoneCount()` to `HealthyInstancesInZoneCount()`. Pay attention that `zonesCount` count logic shouldn't change.

Reply:
Hi @pracucci, 
Actually I was not sure whether we were supposed to count only healthy or all instances in each zone. Thank you for this remark. I have fixed this, and I have also added a test that tests both `HealthyInstancesInZoneCount()` and `ZonesCount()` according to whether the added instance is healthy or not.